File: src/com/quotegraph/interfaces/IVector2I.java
package com.quotegraph.interfaces;

/**
 * Interface for 2D int vectors.
 *
 * @author Daniel Peters
 * @version 1.0
 */
public interface IVector2I {
  int getX();

  int getY();

  void add(IVector2I vector);

  void sub(IVector2I vector);

  void mult(int scalar);

  void div(int scalar);

  int mag();

  void normalize();

  void limit(int max);

  void set(int x, int y);

  void set(IVector2I vector);
}

File: src/main/kotlin/ch/peters/daniel/quotegraph/interfaces/IVector2I.kt
package ch.peters.daniel.quotegraph.interfaces

/**
 * Interface for 2D int vectors.
 *
 * @author Daniel Peters
 * @version 1.0
 */
interface IVector2I {
  var x: Int
  var y: Int

  fun add(vector: IVector2I)

  fun sub(vector: IVector2I)

  fun mult(scalar: Int)

  fun div(scalar: Int)

  fun mag(): Int

  fun normalize()

  fun limit(max: Int)

  fun set(valX: Int, valY: Int)

  fun set(vector: IVector2I)
}
